File: src/main/java/phoenix/init/PhoenixSounds.java
package phoenix.init;

import net.minecraft.util.ResourceLocation;
import net.minecraft.util.SoundEvent;
import net.minecraftforge.fml.RegistryObject;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;
import phoenix.Phoenix;

import java.util.function.Supplier;

public class PhoenixSounds
{
    public static final DeferredRegister<SoundEvent> SOUNDS = new DeferredRegister<>(ForgeRegistries.SOUND_EVENTS, Phoenix.MOD_ID);

    public static final RegistryObject<SoundEvent> UPDATER = SOUNDS.register("updater", sound("oven_fire"));

    public static void register()
    {
        SOUNDS.register(FMLJavaModLoadingContext.get().getModEventBus());
    }

    private static Supplier<SoundEvent> sound(String s)
    {
        return () -> new SoundEvent(new ResourceLocation(Phoenix.MOD_ID, s));
    }
}

File: src/main/kotlin/phoenix/init/PhoenixSounds.kt
package phoenix.init

import net.minecraft.util.ResourceLocation
import net.minecraft.util.SoundEvent
import net.minecraftforge.registries.ForgeRegistries
import phoenix.Phoenix
import phoenix.utils.register
import thedarkcolour.kotlinforforge.forge.KDeferredRegister

object PhoenixSounds
{
    var SOUNDS = KDeferredRegister(ForgeRegistries.SOUND_EVENTS, Phoenix.MOD_ID)

    val CHANGE_STAGE       by SOUNDS.register("change_stage", phoenixSound("change_stage"))
    val PUT_SMTH_TO_BARREL by SOUNDS.register("put_smth_to_barrel", phoenixSound("put_smth_to_barrel"))
    val REDO_MUSIC         by SOUNDS.register("redo_music",  phoenixSound("redo_music"))

    private fun phoenixSound(nameIn: String) = SoundEvent(ResourceLocation(Phoenix.MOD_ID, nameIn))
}